### PR TITLE
Web3/1.0

### LINKED ISSUE #2370: migrate gitcoin from web3js 0.2 to 1.0
the objective of this ticket its to migrate gitcoin from web3js 0.2 to 1.0

i've started a base branch at https://github.com/gitcoinco/web/commit/13bc312dd50b587197932bb7bc8c053b974d68f1 which you can use.

success criteria:
* repo is upgraded to web3js 1.0
* all references to web3.eth.coinbase updated
* BOUNTIES: a funder can post a new bounty
* BOUNTIES: user can fulfill a bounty
* BOUNTIES: funder can increase payout
* BOUNTIES: funder can accept a fulfillment (basic payout)
* BOUNTIES: advanced payout works
* TIPS: a user can post a new tip
* TIPS: a user can receive a tip
* ENS: a user can register an ENS subdomain at /ens
* FAUCET: User can submit a new faucet page
* FAUCET: admin can fund a faucet distribution.
* REQUESTS: User can submit a new requests request

replaces https://github.com/gitcoinco/web/issues/2369

### PR DESCRIPTION
**Pull request by pinkiebell**
Tested:
- [x] onepager
- [x] funding / funding erc20
- [x] bounty details
- [x] ens < until post to backend; backend not configured
- [x] payout
- [x] advanced payout
- [x] cancel bounty
- [x] submit fulfillment
- [x] accept fulfillment
- [x] increase payout
- [x] faucet
- [x] settings
- [x] bounty requests
- [x] kudos

Fixes #2370

### CODE REVIEW

**Comment by pinkiebell** [comment]
@mbeacom Thanks for the changed title, I forgot that to change 😺 .

I need to change a few things back, I will let you know if ready

**Comment by pinkiebell** [comment]
Ready for testing. I will resolve the lint issues afterwards 👍

File: app/assets/v2/js/pages/kudos_details.js [hunk 1]
```diff
@@ -2,13 +2,18 @@ var cloneKudos = function(name, numClones) {
...
+  const account = document.coinbase;
+  const kudosContractInstance = new window.web3.eth.Contract(kudos_abi, kudos_address());
+
+  kudosContractInstance.methods.clone(name, numClones).send(
+    {
+      from: account,
+      value: window.web3.utils.toBN(1000000000000000)
+    },
+    function(error, txid) {
+      console.log('txid:' + txid);
```
[the remark this answers is not in the document]
Comment (octavioamu) [reply app/assets/v2/js/pages/kudos_details.js]: Hey @pinkiebell we can add  /* eslint-disable no-console */ at the top if we need to log the informations so sticler will stop complaining

**Comment by pinkiebell** [comment]
Rebased & ready for review 🎅

**Comment by owocki** [comment]
OK this is deployed at http://kevin.stage.gitcoin.co

**Comment by owocki** [comment]
hmm i get this exception from `sync/web3` when i try to post a new bounty 

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/django/contrib/staticfiles/handlers.py", line 65, in __call__
    return self.application(environ, start_response)
  File "/usr/local/lib/python3.7/site-packages/django/core/handlers/wsgi.py", line 142, in __call__
    response = self.get_response(request)
  File "/usr/local/lib/python3.7/site-packages/django/core/handlers/base.py", line 78, in get_response
    response = self._middleware_chain(request)
  File "/usr/local/lib/python3.7/site-packages/django/core/handlers/exception.py", line 36, in inner
    response = response_for_exception(request, exc)
  File "/usr/local/lib/python3.7/site-packages/django/core/handlers/exception.py", line 90, in response_for_exception
    response = handle_uncaught_exception(request, get_resolver(get_urlconf()), sys.exc_info())
  File "/usr/local/lib/python3.7/site-packages/django/core/handlers/exception.py", line 125, in handle_uncaught_exception
    return debug.technical_500_response(request, *exc_info)
  File "/usr/local/lib/python3.7/site-packages/django_extensions/management/technical_response.py", line 37, in null_technical_500_response
    six.reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.7/site-packages/six.py", line 692, in reraise
    raise value.with_traceback(tb)
  File "/usr/local/lib/python3.7/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.7/site-packages/django/core/handlers/base.py", line 126, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/usr/local/lib/python3.7/site-packages/django/core/handlers/base.py", line 124, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python3.7/site-packages/django/views/decorators/http.py", line 40, in inner
    return func(request, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/ratelimit/decorators.py", line 27, in _wrapped
    increment=True)
  File "/usr/local/lib/python3.7/site-packages/ratelimit/utils.py", line 129, in is_ratelimited
    limited = usage.get('count') > usage.get('limit')
TypeError: '>' not supported between instances of 'NoneType' and 'int'
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): api.github.com:443
```

i think we might have to pull the latest master in, because this is not a prob on the latest master

**Pull request closed without merging**